cpp/include/cudf_test/iterator_utilities.hpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/detail/iterator.cuh>
+#include <cudf/types.hpp>
+
+#include <thrust/iterator/transform_iterator.h>
+
+#include <set>
+
+namespace cudf {
+namespace test {
+
+/**
+ * @brief Bool iterator for marking (possibly multiple) null elements in a column_wrapper.
+ *
+ * The returned iterator yields `false` (to mark `null`) at all the specified indices,
+ * and yields `true` (to mark valid rows) for all other indices. E.g.
+ *
+ * @code
+ * auto iter = iterator_with_null_at(std::vector<size_type>{8,9});
+ * iter[6] == true;  // i.e. Valid row at index 6.
+ * iter[7] == true;  // i.e. Valid row at index 7.
+ * iter[8] == false; // i.e. Invalid row at index 8.
+ * iter[9] == false; // i.e. Invalid row at index 9.
+ * @endcode
+ *
+ * @param indices The collection of indices for which the validity iterator
+ *                must return `false` (i.e. null)
+ * @return auto Validity iterator
+ */
+static auto iterator_with_null_at(std::vector<cudf::size_type> const& indices)
```

Comment:
Using a `std::vector` as the input limits this to only that type. Can the input be made an iterator instead?

Reply:
Nice thing about this interface as-is, is that you can simply call it as `iterator_with_null_at({1,3,5})`. With iterators I think you need to use a variable.
May be better to add an overload instead?

Reply:
Hmm... Actually, yes. @vuule makes a good point.

Permission to leave this as is?

Reply:
Actually, as @vuule advises, we should probably have both.

Reply:
Don't use a vector though, use a `host_span`.